spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -65,18 +63,24 @@ public InputPartition[] planInputPartitions() {
     Broadcast<Table> tableBroadcast = sparkContext.broadcast(SerializableTable.copyOf(table));
     String expectedSchemaString = SchemaParser.toJson(expectedSchema);
 
-    InputPartition[] readTasks = new InputPartition[tasks.size()];
+    InputPartition[] readTasks = new InputPartition[tasks().size()];
```

Comment:
Question: is there any concern with calling `tasks()` repeatedly like this?

Reply:
ja, in some Batch like fileBatch, tasks are recomputed

Reply:
The subclasses all lazily instantiate tasks only once AFAIK